pkg/asset-manager-utils/test/SinglePoolAssetManager.test.ts

```diff
@@ -0,0 +1,211 @@
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { Contract } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+import { encodeJoin } from '@balancer-labs/v2-helpers/src/models/pools/mockPool';
+
+import { bn, fp } from '@balancer-labs/v2-helpers/src/numbers';
+import { MAX_UINT256 } from '@balancer-labs/v2-helpers/src/constants';
+import { GeneralPool } from '@balancer-labs/v2-helpers/src/models/vault/pools';
+
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import { expectBalanceChange } from '@balancer-labs/v2-helpers/src/test/tokenBalance';
+
+const OVER_INVESTMENT_REVERT_REASON = 'investment amount exceeds target';
+
+describe('Single Pool Asset Manager', function () {
+  let tokens: TokenList, vault: Contract, assetManager: Contract;
+  let admin: SignerWithAddress, lp: SignerWithAddress, other: SignerWithAddress;
+  let poolId: string;
+  const tokenInitialBalance = bn(200e18);
+  const amount = bn(100e18);
+
+  before('deploy base contracts', async () => {
+    [admin, lp, other] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('set up asset manager', async () => {
+    // Deploy tokens and vault
+    tokens = await TokenList.create(['DAI', 'MKR'], { sorted: true });
+    const vaultHelper = await Vault.create({ admin });
+    vault = vaultHelper.instance;
+
+    // deploy pool and add liquidity
+    const pool = await deploy('v2-vault/MockPool', { args: [vault.address, GeneralPool] });
+    poolId = await pool.getPoolId();
+
+    // set up asset manager
+    assetManager = await deploy('SinglePoolTestAssetManager', {
+      args: [vault.address, poolId, tokens.DAI.address],
+    });
+
+    await tokens.mint({ to: lp, amount: tokenInitialBalance });
+    await tokens.approve({ to: vault.address, from: [lp] });
+
+    // Assign assetManager to the DAI token, and other to the other token
+    const assetManagers = [assetManager.address, other.address];
+
+    await pool.registerTokens(tokens.addresses, assetManagers);
+
+    await vault.connect(lp).joinPool(poolId, lp.address, other.address, {
+      assets: tokens.addresses,
+      maxAmountsIn: tokens.addresses.map(() => MAX_UINT256),
+      fromInternalBalance: false,
+      userData: encodeJoin(
+        tokens.addresses.map(() => tokenInitialBalance),
+        tokens.addresses.map(() => 0)
+      ),
+    });
+  });
+
+  it('different managers can be set for different tokens', async () => {
+    expect((await vault.getPoolTokenInfo(poolId, tokens.DAI.address)).assetManager).to.equal(assetManager.address);
+    expect((await vault.getPoolTokenInfo(poolId, tokens.MKR.address)).assetManager).to.equal(other.address);
+  });
+
+  it('allows a pool controller to set the desired target investable %', async () => {
```

Comment:
```suggestion
  // TODO: add access control to the Asset Manager
  it.skip('allows a pool controller to set the desired target investable %', async () => {
```

This will make more obvious tthat this test is not really 'done'